AABB_tree/include/CGAL/AABB_tree.h

```diff
@@ -575,7 +578,20 @@ namespace CGAL {
                 const std::size_t range,
                 const ComputeBbox& compute_bbox,
                 const SplitPrimitives& split_primitives,
-                const AABBTraits&);
+                const AABB_traits&);
+
+    template<typename ConstPrimitiveIterator, typename ComputeBbox>
+    void expand(Node &node,
```

Comment:
Why can't you simply rely on the `Split_primitive` nested functor?

Reply:
The template specialization was necessary in order to obtain the performance advantages of construction by sorting. The bulk of its efficiency comes from the fact that the splitting functor doesn't actually need to know the bounding box of the primitives, since it's not choosing what axis to partition them along. This allows the specialized function to calculate bounding boxes after primitives have been distributed, so interior nodes calculate their Bbox by combining the boxes of their children.

This is definitely not a solution I'm happy with though, it breaks some of AABB_tree's established abstractions and required a couple of messy hacks. Let me know if you have a better way of accomplishing the same thing.